Gems/Camera/Code/Source/ViewportCameraSelectorWindow.cpp

```diff
@@ -143,38 +156,38 @@ namespace Camera
 
         //////////////////////////////////////////////////////////////////////////
         /// Maestro::EditorSequenceNotificationBus::Handler
-        void CameraListModel::OnSequenceSelected(const AZ::EntityId& sequenceEntityId)
+        void CameraListModel::OnSequenceSelected(const AZ::EntityId& )
         {
             // Add or Remove the Sequence Camera option if a valid
             // sequence is selected in Track View.
 
             // Check to see if the Sequence Camera option is already present
-            bool found = false;
-            int index = 0;
-            for (const CameraListItem& cameraItem : m_cameraItems)
-            {
-                if (cameraItem.m_cameraName == m_sequenceCameraName)
-                {
-                    found = true;
-                    break;
-                }
-                ++index;
-            }
-
-            // If it is present, but no sequence is selected, removed it.
-            if (found && !sequenceEntityId.IsValid())
-            {
-                beginRemoveRows(QModelIndex(), index, index);
-                m_cameraItems.erase(m_cameraItems.begin() + index);
-                endRemoveRows();
-            }
-            // If it is not present, and there is a sequence selected show it.
-            else if (!found && sequenceEntityId.IsValid())
-            {
-                beginInsertRows(QModelIndex(), rowCount(), rowCount());
-                m_cameraItems.push_back(CameraListItem(m_sequenceCameraName, sequenceEntityId));
-                endInsertRows();
-            }
+            //bool found = false;
```

Comment:
Can you clarify the changes here? Does this remove the sequence camera from the camera selection list entirely, or just remove some logic from selecting it?

Reply:
This change removes all special handling of the "sequence camera". I'm not sure what the "sequence camera" even is, we use track view often but yet I've never seen this mysterious "sequence camera" come into play. Before this change, selecting the "sequence camera" in the list didn't seem to do anything.

Perhaps I should remove all the sequence camera related logic here?

Reply:
I went back and figure out what this whole "sequence camera" business was doing. It was very confusing and I've now removed the functionality entirely.

If you had selected "Sequence Camera" in this list, it wouldn't actually change the camera. But then while "Sequence Camera" is selected, if a track view director node changes the camera, the ViewportCameraSelectorWindow would actually set that camera to be the active camera.

This is really confusing because the "Sequence Camera" in this list isn't a real camera at all, and selecting it doesn't change the active camera. It's a sort of "fake" item that piggypacks on the viewport camera selector widget in order to allow track view camera changes to propagate in editor. This seems very un-intuitive.

In fact, I have just reported the fact that track view director nodes don't apply camera changes in editor as a bug: https://github.com/o3de/o3de/issues/2833 . But all this time I had no idea that selecting "Sequence Camera" would actually give this behavior, because it's such a confusing functionality.